circuits/cpp/src/aztec3/circuits/kernel/private/native_private_kernel_circuit.cpp

```diff
@@ -301,6 +303,42 @@ void validate_inputs(DummyComposer& composer, PrivateInputs<NT> const& private_i
     NT::fr start_public_call_stack_length = array_length(start.public_call_stack);
     NT::fr start_l1_msg_stack_length = array_length(start.l1_msg_stack);
 
+    {
+        // Verify public key hash matches ethereum address of the sender
+        // TODO(Suyash): Do we want to perform this check only for base case?
+        // TODO(Suyash): I think its okay to check this for each kernel iteration.
```

Comment:
It's ok for now, but we can remove it from the 'non initial' kernel circuit, once we write that.